sdk/translation/azure-ai-translation-text/src/main/java/com/azure/ai/translation/text/TextTranslationClientBuilder.java

```diff
@@ -342,6 +357,11 @@ private HttpPipeline createHttpPipeline() {
         policies.add(new AddDatePolicy());
         if (tokenCredential != null) {
             policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPE));
+
+            if (this.region != null && this.azureResourceId != null) {
+                policies.add(new TranslatorRegionAuthenticationPolicy(this.region));
+                policies.add(new TranslatorResourceIdAuthenticationPolicy(this.azureResourceId));
```

Comment:
nit, core actually has https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/AddHeadersPolicy.java

Reply:
Good, I switched to that.